### PR TITLE
feat: Sharing cmc database [CMC-523]

### PR DESCRIPTION
**Pull request by dharmendrak**
Jira: https://tools.hmcts.net/jira/browse/CMC-523

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -215,9 +215,13 @@ dependencies {
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-aop'
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-json'
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-quartz'
+  implementation group: 'org.springframework.boot', name: 'spring-boot-starter-jdbc'
 
   implementation group: 'io.springfox', name: 'springfox-swagger2', version: versions.springfoxSwagger
   implementation group: 'io.springfox', name: 'springfox-swagger-ui', version: versions.springfoxSwagger
+  implementation group: 'org.postgresql', name: 'postgresql', version: '42.2.14'
+  implementation group: 'org.jdbi', name: 'jdbi', version: '2.78'
```
Comment (sabahirfan) [inline build.gradle]: shouldn't we use the version 3 of JDBI rather than version 2?

File: build.gradle [hunk 2]
```diff
@@ -215,9 +215,13 @@ dependencies {
...
+  compile group: 'org.flywaydb', name: 'flyway-core', version: '5.2.4'
```
Comment (sabahirfan) [inline build.gradle]: latest version here as well please `6.5.3`

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -69,6 +74,10 @@ core_case_data:
 spring:
   profiles: local
 
+database:
+  migration:
```
Comment (sabahirfan) [inline src/main/resources/application.yaml:78]: I am not sure whether it is a migration?

**Review by sabahirfan — APPROVED** [review]

File: charts/unspec-service/Chart.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 description: A Helm chart for unspec-service App
 name: unspec-service
 home: https://github.com/hmcts/unspec-service
-version: 1.2.4
+version: 1.2.5
```
Comment (pd-dudkiewicz) [inline charts/unspec-service/Chart.yaml:5]: It should go to 1.3.0 as it's not a bugfix

File: charts/unspec-service/values.preview.template.yaml [hunk 5]
```diff
@@ -1,12 +1,43 @@
...
   ingressHost: ${SERVICE_FQDN}
+  keyVaults:
+    unspec:
+      resourceGroup: unspec
+      secrets:
+        - unspec-idam-client-secret
+        - microservicekey-unspec-service
+        - system-update-user-username
+        - system-update-user-password
+        - AppInsightsInstrumentationKey
```
Comment (pd-dudkiewicz) [inline charts/unspec-service/values.preview.template.yaml:33]: this shouldn't be required as it's in base `values.yaml`

Comment (dharmendrak) [reply charts/unspec-service/values.preview.template.yaml:33]: just to override the vaults config in values.yaml as values.yaml have more specific for env config

Comment (pd-dudkiewicz) [reply charts/unspec-service/values.preview.template.yaml:33]: as discussed, this is required, resolving

**Review by pd-dudkiewicz — APPROVED** [review]

**Pull request merged**